### PR TITLE
[Implementation] Adding get_endpoint functionality to matter_testing support module 

### PR DESCRIPTION
**Pull request by j-ororke**
Incorporating new get_endpoint functionality in matter_testing support module as mentioned in issue #[416](https://github.com/project-chip/matter-test-scripts/issues/416):
- Added prior previous get_endpoint functionality Cecille created to matter_testing support module as mentioned in PR#[31928](https://github.com/project-chip/connectedhomeip/pull/31928)
- Updating test modules to use new get_endpoint functionality

### CODE REVIEW

File: src/python_testing/matter_testing_infrastructure/chip/testing/matter_testing.py
Comment (cecille) [inline src/python_testing/matter_testing_infrastructure/chip/testing/matter_testing.py:1090]: opinion question - do you think the default makes sense here or do you think it makes more sense to throw an error if the test uses an endpoint, but one isn't supplied. 

@raju-apple - thoughts? It might be a bit disruptive, but possibly less error prone?

Either way, let's not block this PR on that.

**Review by cecille — APPROVED** [review]
I'd like to get this in ASAP to avoid conflicts, but would you mind doing a follow-up to update the functions in matter_testing.py to use get_endpoint?

File: src/python_testing/TC_CADMIN_1_11.py
Comment (ReneJosefsen) [inline src/python_testing/TC_CADMIN_1_11.py:31]: Just curious, but this seems like an unrelated change?

File: src/python_testing/TC_DEM_2_10.py
Comment (ReneJosefsen) [inline src/python_testing/TC_DEM_2_10.py:165]: Should we consistently provide a default in all cases?

File: src/python_testing/matter_testing_infrastructure/chip/testing/matter_testing.py
Comment (ReneJosefsen) [reply src/python_testing/matter_testing_infrastructure/chip/testing/matter_testing.py:1090]: I think we should just mandate an endpoint id to be provided, we can do that as a follow up if we do not want to make that change here.

**Review by ReneJosefsen — APPROVED** [review]

File: src/python_testing/TC_DEM_2_10.py
Comment (cecille) [reply src/python_testing/TC_DEM_2_10.py:165]: This matches the current behaviour, so I think for now let's push this through then talk about forced endpoint setting in a followup. That probably needs a bigger PSA / consultation.

File: src/python_testing/TC_CADMIN_1_11.py
Comment (j-ororke) [reply src/python_testing/TC_CADMIN_1_11.py:31]: Hello Rene,
You are correct, this change was needed because the above test TC_CADMIN_1_11 appears to have gotten merged with verbose logging set to enabled for CI test runs which it should be set to disabled. 
Please note that I was unfortunately the one that wrote the above test and appears I forgot to swap verbose logging back to disabled for CI args before the test got merged.

Comment (cecille) [reply src/python_testing/TC_CADMIN_1_11.py:31]: it is, but the change is correct

Comment (cecille) [reply src/python_testing/TC_CADMIN_1_11.py:31]: It just means to not print the output from the test if the test passes. It makes the logs fit in the CI output frame.

File: src/python_testing/matter_testing_infrastructure/chip/testing/matter_testing.py
Comment (cecille) [reply src/python_testing/matter_testing_infrastructure/chip/testing/matter_testing.py:1090]: follow up issue - https://github.com/project-chip/matter-test-scripts/issues/450

**Pull request merged**
